### PR TITLE
feat(envoy): json logging support

### PR DESCRIPTION
**Pull request by byxorna**
## description

This PR adds support to the helm chart to configure envoy with JSON logging via `bootstrap.json`, if `envoy.log.format_json` is provided. This is an alternative to `envoy.log.format`.

I took the opportunity to remove `--log-format` flag from the daemonset, and moved it into the bootstrap config at `application_log_config.log_format.text_format` in order to keep sprawl to a minimum. This change is backwards compatible.

See upstream docs for [envoy application_log_config](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-msg-config-bootstrap-v3-bootstrap-applicationlogconfig)

## checklist

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

### CODE REVIEW

File: install/kubernetes/cilium/files/cilium-envoy/configmap/bootstrap-config.json
Comment (squeed) [inline install/kubernetes/cilium/files/cilium-envoy/configmap/bootstrap-config.json]: nit: this comma is extra :-)

Comment (byxorna) [reply install/kubernetes/cilium/files/cilium-envoy/configmap/bootstrap-config.json]: Good catch, thanks! updated in https://github.com/cilium/cilium/commit/fd54b009cd483ff1ad1c68ff11da4fee4b53d779

**Comment by squeed** [comment]
Looks reasonable to me. I'll let @sayboras comment from the Envoy side.

File: USERS.md
Comment (sayboras) [inline USERS.md]: we would love to have this in separate PR

**Review by sayboras — COMMENTED** [review]
The changes look reasonable to me. Can you split the changes for USERS.md to another PR?

File: USERS.md
Comment (byxorna) [reply USERS.md]: Good suggestion, I opened https://github.com/cilium/cilium/pull/34603 👍🏽

**Review by sayboras — APPROVED** [review]
Thanks and lgtm :heavy_check_mark:

**Comment by byxorna** [comment]
@sayboras @squeed 👋🏽  - how can i assist to get this PR merged? Looks like there are some unrelated CI failures (ccache missing from some jobs?), not sure if those are blockers. Thanks! 😄

**Comment by julianwiedmann** [comment]
@byxorna please don't merge the `main` branch into your PR. Just rebase instead :).

**Comment by byxorna** [comment]
> @byxorna please don't merge the `main` branch into your PR. Just rebase instead :).

Sorry about that! i just rebased and pushed, will do that next time

**Comment by julianwiedmann** [comment]
@squeed sorry, this still needs a review for your codeowners.

**Review by squeed — APPROVED** [review]

**Pull request merged**
